### ISSUE TITLE
Parse comments into AST

### ISSUE THREAD

**Issue by certik** [lfortran/lfortran#255]
*Original issue*: https://gitlab.com/lfortran/lfortran/-/issues/173

And translate them in the C++ translation.

**Comment by lucifer1004** [comment]
Comments are now in AST, but not in ASR, so the first part is solved but the second part is still a to-do.

**Comment by certik** [comment]
Yes, indeed. The other issue is that the parser stores the comments *after* each statement, but it is much more natural to store them before each statement, but we have not figured out how to do it in the parser. So the AST->ASR should reattach the comments correctly.
